addons/website/models/website_page.py

```diff
@@ -340,6 +341,13 @@ def _search_fetch(self, search_detail, search, limit, order):
                 )
 
         def filter_page(search, page, all_pages):
+            # Exclude pages that do not pass ACL.
+            try:
+                plain_page = page.sudo(False)
+                plain_page.check_access_rule('read')
+                plain_page.view_id.check_access_rule('read')
+            except AccessError:
+                return False
```

Comment:
I am a bit confused by the need of `sudo` in the code surrounding this. We are only *searching* for results (not updating anything), if we end up filtering on records where we do have read access... why are we not searching for those results without sudo in the first place? :thinking:

I actually don't understand the `requires_sudo` logic around this as well.

Reply:
The whole `requires_sudo` mess is because public user is not allowed to read `website.page`. (And therefore we need to "un-sudo" here.)

E.g.
```py
>>> env['website.page'].with_user(env.ref("base.public_user").id).search([])
2024-06-21 11:19:22,793 226049 INFO c odoo.addons.base.models.ir_model: Access Denied by ACLs for operation: read, uid: 4, model: website.page 
Traceback (most recent call last):
  File "/home/odoo/src/odoo/odoo/tools/cache.py", line 85, in lookup
    r = d[key]
  File "/home/odoo/src/odoo/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('ir.model.access', <function IrModelAccess.check at 0x7f3a7c1e0280>, 4, False, 'website.page', 'read', True, ('en_US',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<console>", line 1, in <module>
  File "/home/odoo/src/odoo/odoo/models.py", line 1821, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/home/odoo/src/odoo/odoo/models.py", line 4730, in _search
    model.check_access_rights('read')
  File "/home/odoo/src/odoo/odoo/models.py", line 3563, in check_access_rights
    return self.env['ir.model.access'].check(self._name, operation, raise_exception)
  File "<decorator-gen-35>", line 2, in check
  File "/home/odoo/src/odoo/odoo/tools/cache.py", line 90, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_model.py", line 1851, in check
    raise AccessError(msg)
odoo.exceptions.AccessError: You are not allowed to access 'Page' (website.page) records.

This operation is allowed for the following groups:
	- Website/Editor and Designer

Contact your administrator to request access if necessary.
```